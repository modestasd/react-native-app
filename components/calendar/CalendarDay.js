import * as React from 'react';
import { Text, View, StyleSheet,TouchableOpacity,Dimensions } from 'react-native';

const CalendarDay = ({dayName, day, onDayClick,isActive, index}) => {
  return ( 
    <TouchableOpacity style={styles.calendarDays} onPress={() => onDayClick(index, day)}>
      <View style={{ flex: 1, alignItems: 'center' }}>
        <Text style={styles.calendarDayName}>{dayName}</Text>
        <View style={isActive ? styles.activeCalendarDayContainer : styles.calendarDayContainer}>
          <Text style={isActive ? styles.activeCalendarDayNumber : styles.calendarDayNumber}>{day}</Text>
        </View>
      </View>
    </TouchableOpacity >
  );
};

export default CalendarDay;

const styles = StyleSheet.create({
  calendarDays: {
    width: 60,
  },
  calendarDayName: {
    fontSize: 16,
    color: 'grey',
    marginBottom: 5,
    height: 20
  },
  calendarDayNumber: {
    fontSize: 18,
    fontWeight: '600',
    height: 40,
    textAlign: 'center'
  }, 
  calendarDayContainer:{
    height: 40,
    width: '100%',
    padding: 5,
  },
  activeCalendarDayContainer: {
    backgroundColor: '#4169e1',
    height: 35,
    width: '60%',
    padding: 5,
    borderRadius: 50,
  },
  activeCalendarDayNumber: { 
    fontSize: 18,
    fontWeight: '600',
    color: 'white',
    textAlign: 'center'
  }
})