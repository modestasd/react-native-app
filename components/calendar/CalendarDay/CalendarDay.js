import React from 'react';
import PropTypes from 'prop-types';

import {CalendarDayWrapper,DayName,Day, DayWrapper} from './CalenndarDay.styles';

const CalendarDay = ({dayName, day, onDayClick, isActive,index}) => {
  return ( 
    <CalendarDayWrapper onPress={() => onDayClick(index, day)}>
      <DayName>{dayName}</DayName>
      <DayWrapper isActive={isActive}>
        <Day isActive={isActive}>{day}</Day>
      </DayWrapper>
    </CalendarDayWrapper >
  );
};

CalendarDay.propTypes = {
	dayName: PropTypes.string.isRequired,
  day: PropTypes.number.isRequired,
	onDayClick: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
  index: PropTypes.number.isRequired,
};

CalendarDay.defaultProps = {
	isActive: false
};
 
export default CalendarDay;
