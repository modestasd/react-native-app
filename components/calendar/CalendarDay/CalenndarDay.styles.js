import styled from 'styled-components/native';

export const CalendarDayWrapper = styled.TouchableOpacity`
    width: 60;
    align-items: center;
    flex: 1;
`;

export const DayName = styled.Text`
    font-size: 16;
    color: grey;
    margin-bottom: 5;
    height: 20;
`;

export const DayWrapper = styled.View`
    height: ${({isActive}) => isActive ? "35" : "40"};
    width: ${({isActive}) => isActive ? "60%" : "100%"};
    padding: 5px;
    background-color: ${({isActive}) => isActive ? "#4169e1" : 'white'}; 
    border-radius: ${({isActive}) => isActive ? "50px" : ''}; 
`; 

export const Day = styled.Text`
    font-size: 18;
    font-weight: bold;
    height: 40;
    text-align: center;
    color: ${({isActive}) => isActive ? "white" : "black"}; 
`; 