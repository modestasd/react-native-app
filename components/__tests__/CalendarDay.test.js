import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components/native'
import { find } from 'styled-components/test-utils'

import CalendarDay from '../calendar/CalendarDay';
import {CalendarDayWrapper,DayName,Day, DayWrapper} from '../calendar/CalendarDay/CalenndarDay.styles';

describe('CalendarDay', () => {
    it('expect to match latest snapshot', () => {
        const wrapped = renderer.create(
            <CalendarDay 
                dayName='Pirm' 
                day={1} 
                onDayClick={()=> console.log('Test')} 
                isActive={true} index={2}
            />
        ).toJSON();
        expect(wrapped).toMatchSnapshot(); 
    });
});