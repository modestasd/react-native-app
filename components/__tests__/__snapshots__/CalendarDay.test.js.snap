// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay expect to match latest snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "opacity": 1,
      "width": 60,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "grey",
          "fontSize": 16,
          "height": 20,
          "marginBottom": 5,
        },
      ]
    }
  >
    Pirm
  </Text>
  <View
    isActive={true}
    style={
      Array [
        Object {
          "backgroundColor": "#4169e1",
          "borderRadius": 50,
          "height": 35,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
          "width": "60%",
        },
      ]
    }
  >
    <Text
      isActive={true}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
            "height": 40,
            "textAlign": "center",
          },
        ]
      }
    >
      1
    </Text>
  </View>
</View>
`;

exports[`CalendarDay has 1 child 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "opacity": 1,
      "width": 60,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "grey",
          "fontSize": 16,
          "height": 20,
          "marginBottom": 5,
        },
      ]
    }
  >
    Pirm
  </Text>
  <View
    isActive={true}
    style={
      Array [
        Object {
          "backgroundColor": "#4169e1",
          "borderRadius": 50,
          "height": 35,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
          "width": "60%",
        },
      ]
    }
  >
    <Text
      isActive={true}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
            "height": 40,
            "textAlign": "center",
          },
        ]
      }
    >
      1
    </Text>
  </View>
</View>
`;
